Automattic/simplenote-ios
# Issue/517 share to books

### Fix
This is a PR to fix #517 

The issue in #517  is that sharing from Simplenote to Apple Books is an option in the share sheet, but it doesn't work.  A message appears saying that it is converting the note to a PDF but nothing ever happens.

In this PR we are creating the PDF for Books instead and then supplying the ActivityView a link to where the file can be found so books can open the link.

This is a proof of concept about how we could potentially fix this problem, looking for some feedback on the approach

### Test
To test:
1. Open up a note and go to Options > Share > Books (you may have to look for it in the more options)
2. books should open and a PDF copy of your note should appear

A few things to keep in mind.  
-This is not styled in any way at this time, and we may need to more carefully consider what the PDF output should be (markup?  text styles?  etc)
-This should be able to continue creating new versions of PDFs if the note has changes to it, if there are no changes it should open the existing file
-This should only trigger the creation of the PDF once, at the time that share to books is tapped, but would be good to confirm.
-Making the PDF takes a bit of time, 2-3 seconds, but this happens while you are looking at books, so while it feels a bit laggy, it doesn't feel like it is laggy on SNs end

What do you think about how this is setup?
### Review
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.
